src/programs/main.c

```diff
@@ -0,0 +1,28 @@
+#include <environment.h>
+#include <nongnu/unistd.h>
+#include <string.h>
+#include <drivers/kinetis/uart.h>
+#include <stdint.h>
+#include <kernel/process.h>
+#include <kernel/stream_io.h>
+#include <kernel/private/static_memory.h>
+#include <kernel/schedule.h>
+#include <drivers/arm/cm4/systick.h>
+#include <programs/sensor_read.h>
+#include <drivers/devices/status_leds.h>
+
+/**
+ * Entry point for the process switcher.
+ */
+int init_main(void) {
+#ifdef TASK_EN_SENSOR_READ
+    process_init(&sensor_read_app, &sensor_read_main, sensor_read_stack,
+                 SENSOR_READ_STACK_SIZE, true, false);
+    schedule_process(&process_table, &sensor_read_app);
+#endif
+
+#if !defined(TASK_EN_SENSOR_READ)
```

Comment:
What I'm going for here is a way to prevent a hardfault for accidentally having an empty main. Maybe we can light an LED or something to denote that you have nothing in your main?

Reply:
Conceptually, I don't think we should attempt to prevent hardfaults.  If we wanted to deadloop even if no processes are started, I can think of a few ways to do it:
 - in the process  switcher, deadloop if there are no running processes.  This is quick, but has the disadvantage of locking up the system in a weird spot.
 - in kernel_main, only start systick if processes were created.
 - Start systick in the init_main process only if there were processes scheduled
 - Implement the fork system call to start new processes, thus preventing anything from happening unless the init process forks.